### PR TITLE
CIRCLE-16271 apply patch for CVE 2019-5736

### PR DESCRIPTION
**Pull request by bear**
Adjust the cloud init script that installs Docker for CVE 2019-5736

This change uses jq to add a single entry to the installed docker config file

Doesn't assume the presence of jq
Saves the original file to /etc/docker/daemon.json.orig
Assumes:
* the path for the docker config is /etc/docker/daemon.json
* we are able to write to the above file

### CODE REVIEW

File: provision-nomad-client-ubuntu.sh [hunk 1]
```diff
@@ -80,6 +80,12 @@ else
 fi
 apt-get -y install docker-ce=$(docker_package_name)
 
+# force docker to use userns-remap to mitigate CVE 2019-5736
+apt-get -y install jq
+tmp=$(mktemp)
+cp /etc/docker/daemon.json /etc/docker/daemon.json.orig
+jq '."userns-remap": "default"' /etc/docker/daemon.json > "$tmp" && mv "$tmp" /etc/docker/daemon.json
```
Comment (mharriscircleci) [inline provision-nomad-client-ubuntu.sh]: i'm getting a syntax error testing this jq statement on my machine.   The assignment operator in jq is `=`

File: provision-nomad-client-ubuntu.sh [hunk 2]
```diff
@@ -80,6 +80,12 @@ else
 fi
 apt-get -y install docker-ce=$(docker_package_name)
 
+# force docker to use userns-remap to mitigate CVE 2019-5736
+apt-get -y install jq
+tmp=$(mktemp)
+cp /etc/docker/daemon.json /etc/docker/daemon.json.orig
```
Comment (dcarley) [inline provision-nomad-client-ubuntu.sh]: I hacked circleci/server-manifest to test this:
- circleci/server-manifest@755cff6a7907141a05d7cf39c43ec49dc6580a4f
- https://circleci.com/gh/circleci/server-manifest/23483

It looks like this file doesn't exist by default:
```
[ 2019-02-13 11:01:14,338 ] ERROR - + cp /etc/docker/daemon.json /etc/docker/daemon.json.orig
[ 2019-02-13 11:01:14,338 ] ERROR - cp: cannot stat ‘/etc/docker/daemon.json’: No such file or directory
```

Maybe we should create it with an empty hash if it doesn't exist?

File: provision-nomad-client-ubuntu.sh (hunk 1 above)
Comment (bear) [reply provision-nomad-client-ubuntu.sh]: fixed

File: provision-nomad-client-ubuntu.sh (hunk 2 above)
Comment (bear) [reply provision-nomad-client-ubuntu.sh]: code was added to push an empty hash if it doesn't exist

**Comment by shamshad95** [comment]
Hi

**Pull request merged**